odoo/tools/pdf/__init__.py

```diff
@@ -217,6 +217,24 @@ def to_pdf_stream(attachment) -> io.BytesIO:
     _logger.warning("mimetype (%s) not recognized for %s", attachment.mimetype, attachment)
 
 
+def extract_page(attachment, num_page=0) -> io.BytesIO | None:
+    """Exctract a specific page form an attachement pdf"""
+    pdf_stream = to_pdf_stream(attachment)
+    if not pdf_stream:
+        return
+    try:
+        pdf = PdfFileReader(pdf_stream)
+    except PdfReadError:
+        _logger.warning("Error reading pdf file %s", attachment)
```

Comment:
Should it be a warning? Can server admin do anything about it? Probably not.

Reply:
I don't want this workaround, this is ugly and makes no sense.

Either we shouldn't return html, or the mimetype should be correct so we don't even attempt to parse that as a pdf.

When we get a failing pdf, we should have an error when trying to generate a thumbnail from it. I suggested to catch the error in the controller and raise it as a user error instead.

Reply:
> I suggested to catch the error in the controller and raise it as a user error instead.

This solution creates a warning that makes the runbot red.
I feel like I'm running in circles,